packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -704,13 +708,17 @@ abstract class State<T extends StatefulWidget> {
   /// The framework will call this method exactly once for each [State] object
   /// it creates.
   ///
+  /// You cannot use [BuildContext.inheritFromWidgetOfExactType] from this
+  /// method. However, [dependenciesChanged] will be called immediately
+  /// following this method, and [BuildContext.inheritFromWidgetOfExactType] can
+  /// be used there.
+  ///
   /// If you override this, make sure your method starts with a call to
   /// super.initState().
   @protected
   @mustCallSuper
   void initState() {
     assert(_debugLifecycleState == _StateLifecycle.created);
-    assert(() { _debugLifecycleState = _StateLifecycle.initialized; return true; });
```

Comment:
We used to have this assignment here to ensure that initState was called all the way up the inheritance tree.  This patch loses that, but maybe that doesn't matter now that `@mustCallSuper` enforces that property statically?